backend/routes/opportunity.js

```diff
@@ -0,0 +1,288 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const nodemailer = require("nodemailer");
+const moment = require("moment");
+
+const app = express();
+const Opportunity = require("../models/opportunity");
+const Volunteer = require("../models/volunteer");
+
+const transport = nodemailer.createTransport({
+  service: "gmail",
+  auth: {
+    user: `${process.env.EMAIL_USER}`,
+    pass: `${process.env.EMAIL_PASSWORD}`,
+  },
+  logger: true,
+});
+
+const postNewOpportunity = async (
+  title,
+  description,
+  pictures,
+  startEvent,
+  endEvent,
+  skills,
+  wishlist,
+  location,
+  requirements,
+  tasks,
+  additionalInfo,
+  volunteers
+) =>
+  new Opportunity({
+    title,
+    description,
+    pictures,
+    startEvent,
+    endEvent,
+    skills,
+    wishlist,
+    location,
+    requirements,
+    tasks,
+    additionalInfo,
+    volunteers,
+  }).save();
+
+app.get("/api/getOpportunity/:id", async (req, res) => {
+  const oppid = mongoose.Types.ObjectId(req.params.id);
+  try {
+    const opp = await Opportunity.findById(oppid);
+    console.log(opp);
+    res.status(200).json(opp);
+  } catch (error) {
+    console.log(error);
+    res.status(400).send(error);
+  }
+});
+
+app.get("/api/opportunityDetail/:name", async (req, res) => {
+  const name = mongoose.Types.ObjectId(req.params.name);
+  try {
+    const opp = await Opportunity.findById(name);
+    res.status(200).json(opp);
+  } catch (error) {
+    console.log(error);
+    res.status(400).send(error);
+  }
+});
+
+app.get("/api/opportunities", async (req, res) => {
+  try {
+    const opp = await Opportunity.find({});
+    res.status(200).json(opp);
+  } catch (error) {
+    console.log(error);
+    res.status(400).send(error);
+  }
+});
+
+// const getAllOpportunities = async () => Opportunity.find({});
+
+app.post("/api/updateOpportunity", async (req, res) => {
+  try {
+    if (typeof req.body._id === "undefined" || req.body._id === "") {
```

Comment:
This should be `req.body._id === undefined` instead of `typeof req.body._id === "undefined"` because we don't want to check the type